plugin/entryAttributes.go

```diff
@@ -0,0 +1,181 @@
+package plugin
+
+import (
+	"encoding/json"
+	"os"
+	"time"
+
+	"github.com/Benchkram/errz"
+)
+
+// EntryMetadata represents the entry's metadata
+// as returned by Entry#Metadata.
+type EntryMetadata map[string]interface{}
+
+// ToMeta converts an object to an EntryMetadata object. If the input is already an
+// array of bytes, it must contain a serialized JSON object. Will panic if given something
+// besides a struct or []byte.
+func ToMeta(obj interface{}) EntryMetadata {
+	var err error
+	var inrec []byte
+	if arr, ok := obj.([]byte); ok {
+		inrec = arr
+	} else {
+		if inrec, err = json.Marshal(obj); err != nil {
+			// Internal error if we can't marshal an object
+			panic(err)
+		}
+	}
+	var meta EntryMetadata
+	// Internal error if not a JSON object
+	errz.Fatal(json.Unmarshal(inrec, &meta))
+	return meta
+}
+
+// EntryAttributes represents an entry's attributes. We use a struct
+// instead of a map for efficient memory allocation/deallocation,
+// which is needed to make Group#List fast.
+type EntryAttributes struct {
```

Comment:
I wonder if we should include the entry's supported actions here via an `actions` attribute. Could fill this info. in `CachedList`.

Reply:
What benefit would that bring? Where do we currently do it?

Reply:
Was thinking symmetry reasons (`actions` really are an attribute) but now that you mention it, it could be useful as a quick optimization to the `[`IsSupportedOn` methods](https://github.com/puppetlabs/wash/blob/master/plugin/action.go#L16-L26)` b/c we can just calculate that once in `CachedList` and indefinitely store it.